osc/core.py

```diff
@@ -4844,7 +4865,11 @@ def get_source_file_diff(dir, filename, rev, oldfilename = None, olddir = None,
             d[i] += b'\n\\ No newline at end of file'
             if i+1 != len(d):
                 d[i] += b'\n'
-    return d
+
+    if COLORIZE:
```

Comment:
needs a check for whether it is connected to a terminal or not (isatty)

Reply:
Thank you! Fixed.